### PR TITLE
Update header content in Direct Schedule workflow

### PR DESCRIPTION
**Pull request by JunTaoLuo**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- This PR updates the headers for the Direct scheduling flow pages according to the design file [here](https://www.figma.com/design/5dWfMQkBCvEX9XYJsNdcT3/Change-direct-schedule-headings-to-questions-%7C-Appointments-FE?node-id=1-54&node-type=canvas&t=CsopUwqGZnkaeKBc-0).
  - [x] Type of Care
  - [x] Type of Facility
  - [x] VA Facility
  - [x] Clinic Choice
  - [x] Preferred Date
  - [x] Date and Time
  - [x] Reason for Appointment
- Appointments (VAOS) Team
- This work is not behind a Flipper toggle.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/95036

## Testing done

- Updated unit tests to accommodate the new headers
- Tested manually, see screenshots below

### Desktop

I'm including desktop screenshots here since I had to add customized styling for content width when using radio fields. For future reference, consider this documentation for why the customization was added in this PR.

| DS Step | Before | After |
| ------- | ------ | ----- |
| Type of Care  |  image | image |
| Type of Facility  |  image | image |
| Clinic Choice - Single  |  image | image |
| Clinic Choice - Multiple  |  image |  image |
| Reason for Appointment  |  image |  image |

### Mobile

|    DS Step     | Before | After |
| ------- | ------ | ----- |
| Type of Care  |  image     |     image |
| Type of Facility |   image   | image     |
| VA Facility |   image    |  image  |
| Clinic Choice - Single |  image   |  image  |
| Clinic Choice - Multiple |   image    |   image   |
| Preferred Date |   image    |  image    |
| Date and Time |   image    |  image |
| Reason for Appointment |   image    |  image  |

## What areas of the site does it impact?

Appointments (VAOS)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/vaos/tests/e2e/page-objects/ClinicChoicePageObject.js [hunk 1]
```diff
@@ -17,13 +16,15 @@ export class ClinicChoicePageObject extends PageObject {
   selectClinic({ selection, isCovid = false } = {}) {
     if (isCovid) {
       cy.findByText(/Choose a clinic/i, { selector: 'h1' });
+      cy.findByLabelText(selection).as('radio');
+      cy.get('@radio').check();
     } else {
-      cy.findByText(/Choose a VA clinic/i, { selector: 'h1' });
+      cy.get('va-radio')
+        .shadow()
+        .get(`va-radio-option[value=${selection}]`)
```
Comment (JunTaoLuo) [inline src/applications/vaos/tests/e2e/page-objects/ClinicChoicePageObject.js:24]: A little odd but it seems that selection by text works inconsistently with VARadioField so I'm opting to select by value instead, which works more consistently.

File: src/applications/vaos/tests/mocks/setup.js [hunk 2]
```diff
@@ -306,19 +306,22 @@ export async function setVaccineFacility(store, facilityId, facilityData = {}) {
...
     <Route component={ClinicChoicePage} />,
     {
       store,
     },
   );
 
-  fireEvent.click(await screen.findByLabelText(label));
+  const radioSelector = screen.container.querySelector('va-radio');
+  const changeEvent = new CustomEvent('selected', {
+    detail: { value },
```
Comment (JunTaoLuo) [inline src/applications/vaos/tests/mocks/setup.js:322]: Similar to the E2E tests, I'm opting to select by value which works more consistently.

**Comment by JunTaoLuo** [comment]
FYI @outerpress I couldn't get the format on the "Type of facility" page to match the design exactly. As far as I can tell, the `VaRadioField` component doesn't allow us to format the option labels so I can't apply bold font and change the size. Let me know if that's an issue and I might look into writing our own radio widget for it. I might also ask @vbahinwillit and see if he has any other ideas for it. Otherwise the rest of the pages are good to go.

**Comment by outerpress** [comment]
@JunTaoLuo it looks like the Tile version is the only one in the design system that includes descriptive text: https://design.va.gov/storybook/?path=/story/uswds-va-radio--tile

Is that something we could implement here? Separate ticket is fine if needed, as long as we can do the follow-up work soon after this goes out.

**Comment by JunTaoLuo** [comment]
@outerpress I wasn't aware of that component so let me play around with it a bit. I'll follow up with Tony and Jennie next week as well to see if there's a better approach.

**Comment by outerpress** [comment]
@JunTaoLuo that's weird,the design system docs make it sound like it's intended to show more info:

> Use the [Tile](https://design.va.gov/components/form/radio-button#tile) variation to provide additional details about one or more radio button options within a large and well defined tap target.

Also in the screenshot you posted the alignment of the radio button and font size look different from what's in Storybook. Could there be other configuration options? Or maybe I'm misreading.

**Comment by JunTaoLuo** [comment]
From my testing, it seems like tile variation has no impact on whether descriptions are displayed. As for the alignment, it turns out that we have [customized styling](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/vaos/new-appointment/sass/styles.scss#L15-L18):

| Default styles | VAOS customized styles |
| - | - |
| image | image |

This does give me an idea of modifying the text via some kind of css voodoo, which I've wary of.

File: src/applications/vaos/tests/e2e/page-objects/TypeOfCarePageObject.js [hunk 3]
```diff
@@ -20,17 +20,17 @@ export class TypeOfCarePageObject extends PageObject {
...
 
   selectTypeOfCare(label) {
-    cy.findByLabelText(label)
-      .as('radio')
-      .focus();
-    cy.get('@radio').check();
+    cy.get('va-radio')
+      .shadow()
+      .get(`va-radio-option[value=${label}]`)
+      .click();
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/e2e/page-objects/TypeOfCarePageObject.js:33]: Why the change? Ideally, it's better to write test by what the user sees on the screen as opposed to implementation details that might change.

File: src/applications/vaos/tests/e2e/page-objects/TypeOfFacilityPageObject.js [hunk 4]
```diff
@@ -10,10 +10,10 @@ export class TypeOfFacilityPageObject extends PageObject {
   }
 
   selectTypeOfFacility(label) {
-    cy.findByLabelText(label)
-      .as('radio')
-      .focus();
-    cy.get('@radio').check();
+    cy.get('va-radio')
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/e2e/page-objects/TypeOfFacilityPageObject.js:13]: Same question as above.

File: src/applications/vaos/tests/e2e/page-objects/TypeOfCarePageObject.js (hunk 3 above)
Comment (JunTaoLuo) [reply src/applications/vaos/tests/e2e/page-objects/TypeOfCarePageObject.js:33]: `findByLabelText` doesn't work with the `VaRadioField` widget, I didn't investigate much but it might have something to do with the shadow DOM. I took a look at how others wrote tests when using the `VaRadioField` and this is how they interact with it. Selecting by value also works consistently across E2E test and unit tests.

File: src/applications/vaos/tests/e2e/page-objects/ClinicChoicePageObject.js (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/e2e/page-objects/ClinicChoicePageObject.js:24]: I would look into this a little more since checking by value is an implementation detail. It's recommended to test what is seen on the screen.

Comment (JunTaoLuo) [reply src/applications/vaos/tests/e2e/page-objects/ClinicChoicePageObject.js:24]: Hmm okay let me try a few more things.

**Review by vbahinwillit — CHANGES_REQUESTED** [review]
I noticed both the unit and e2e tests have change to testing implementation details for type of care as opposed to what is seen on the screen. Please see [React Testing Library](https://testing-library.com/docs/) for more information on why this is not considered a good practice.

**Comment by JunTaoLuo** [comment]
@vbahinwillit The e2e tests have been updated, thanks for calling that out. Unfortunately unit tests don't have access to the shadow DOM so they will have to rely on setting the value directly.

File: src/applications/vaos/new-appointment/components/TypeOfFacilityPage.jsx [hunk 5]
```diff
@@ -23,50 +25,39 @@ const initialSchema = {
...
+      'ui:options': {
+        classNames: 'vads-u-margin-top--neg2',
+        showFieldLabel: false,
+        labelHeaderLevel: '1',
+        descriptions: {
+          [FACILITY_TYPES.VAMC]:
+            'Go to a VA medical center or clinic for this appointment',
+          [FACILITY_TYPES.COMMUNITY_CARE]:
+            'Go to a community care facility near your home',
+        },
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/components/TypeOfFacilityPage.jsx:56]: I'm wondering if having the label placed within the ui:options just above the descriptions can provide greater context.
```suggestion
      'ui:options': {
        classNames: 'vads-u-margin-top--neg2',
        showFieldLabel: true,
        labelHeaderLevel: '1',
        labels: {
          [FACILITY_TYPES.VAMC]: 'VA medical center or clinic',
          [FACILITY_TYPES.COMMUNITY_CARE]: 'Community care facility',
        },
        descriptions: {
          [FACILITY_TYPES.VAMC]:
            'Newer text from figma file',
          [FACILITY_TYPES.COMMUNITY_CARE]:
            'Newer text from figma file',
        },
```

**Review by jenniemc — CHANGES_REQUESTED** [review]
Based on the [figma file](https://www.figma.com/design/5dWfMQkBCvEX9XYJsNdcT3/Change-direct-schedule-headings-to-questions-%7C-Appointments-FE?node-id=44-8193&node-type=instance&t=IessRJo5rvopo3cv-0), looks like there's a new set of text for the descriptive section of `TypeOfFacilityPage` (Choose VA or CC) radio button

File: src/applications/vaos/new-appointment/components/TypeOfFacilityPage.jsx (hunk 5 above)
Comment (JunTaoLuo) [reply src/applications/vaos/new-appointment/components/TypeOfFacilityPage.jsx:56]: Ah that looks much better.

**Comment by jenniemc** [comment]
@outerpress , Previously in the [Choose Clinic flow page](https://www.figma.com/design/ugE1APC20v8OcArGB2IMQy/User-Flows-%7C-Appointments-FE?node-id=1-3301&node-type=section&t=FOigxSYhJEO3BhHZ-0), there are textual distinctions between single clinic versus multiple clinics. Will there be such textual description on single clinic with this update?

**Comment by outerpress** [comment]
@jenniemc I dropped a screenshot of the current single clinic page and yes we should keep the text/question I highlighted in the box. If this is an issue with the current implementation lmk:

Screenshot 2024-11-04 at 2 38 31 PM

**Comment by jenniemc** [comment]
@JunTaoLuo, Looks like we're keeping the descriptive text for single clinic. See Peter's comments above.

**Comment by JunTaoLuo** [comment]
Hmm now I'm confused, if we're keeping all the text inside the box, then what should the header be? Specifically, what should "Choose a VA clinic" be changed to?

**Review by jenniemc — APPROVED** [review]
This looks terrific!

**Review by ataker — APPROVED** [review]

**Review by vbahinwillit — APPROVED** [review]
LGTM!. Thanks for making the e2e changes.
